### PR TITLE
Add settings for custom bridges

### PR DESCRIPTION
**Pull request by grote**
Adds settings to set custom bridges to Tor. These are only shown in debug builds, because they are not actually functional. This PR is just about the UI. Making use of them in `TorManager` will be done separately.

First part of #39

### CODE REVIEW

File: app/src/main/java/org/onionshare/android/ui/settings/SettingsTorUi.kt [hunk 1]
```diff
@@ -0,0 +1,287 @@
...
+    Scaffold(
+        scaffoldState = scaffoldState,
+        topBar = {
+            AboutActionBar(navController, R.string.settings_tor_title)
+        }
+    ) { innerPadding ->
+        SettingsTorUiContent(
+            country = settingsManager.currentCountry,
+            automaticBridges = settingsManager.automaticBridges.value,
+            numberOfCustomBridges = settingsManager.customBridges.value.size,
```
Comment (akwizgran) [inline app/src/main/java/org/onionshare/android/ui/settings/SettingsTorUi.kt:79]: With LiveData there's a risk when calling getValue() that the value may not have been initialised yet (if it wasn't initialised in the constructor). Does the same risk exist for State, or is it guaranteed that the value will have been loaded from shared preferences before getValue() returns?

File: app/src/main/java/org/onionshare/android/ui/settings/SettingsTorUi.kt [hunk 2]
```diff
@@ -0,0 +1,287 @@
...
+) {
+    val scrollableState = rememberScrollState()
+    Column(modifier = modifier.verticalScroll(scrollableState)
+    ) {
+        Text(
+            modifier = Modifier
+                .padding(horizontal = 16.dp)
+                .padding(top = 16.dp, bottom = 8.dp),
+            text = stringResource(R.string.settings_tor_intro) + " " +
+                stringResource(R.string.settings_tor_automatic_info),
```
Comment (akwizgran) [inline app/src/main/java/org/onionshare/android/ui/settings/SettingsTorUi.kt:114]: Looks like these strings are only used here - could they be merged?

File: app/src/main/java/org/onionshare/android/ui/settings/SettingsTorUi.kt [hunk 3]
```diff
@@ -0,0 +1,287 @@
...
+    title: String,
+    summary: String,
+    selected: Boolean,
+    onSelected: () -> Unit,
+) {
+    Row(
+        modifier = Modifier
+            .fillMaxWidth()
+            .selectable(
+                enabled = !selected,
```
Comment (akwizgran) [inline app/src/main/java/org/onionshare/android/ui/settings/SettingsTorUi.kt:240]: Is this disabling the radio button when it's not selected?

**Review by akwizgran — APPROVED** [review]
Looks good!

File: app/src/main/java/org/onionshare/android/ui/settings/SettingsTorUi.kt (hunk 3 above)
Comment (grote) [reply app/src/main/java/org/onionshare/android/ui/settings/SettingsTorUi.kt:240]: Unfortunately not the radio button, but the text label of the button (making it non-selectable). I tried also disabling the radio button, but that shows visually which is not good.

File: app/src/main/java/org/onionshare/android/ui/settings/SettingsUi.kt [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+
+@Composable
+fun Preference(
+    title: String,
+    summary: String,
+    onClick: () -> Unit,
+) {
+    Column(modifier = Modifier
+        .fillMaxWidth()
+        .clickable { onClick() }
```
[the remark this answers is not in the document]
Comment (grote) [reply app/src/main/java/org/onionshare/android/ui/settings/SettingsUi.kt:64]: Right, we can also do:

```kotlin
.clickable(onClick = onClick)
```

File: app/src/main/java/org/onionshare/android/ui/settings/SettingsTorUi.kt (hunk 1 above)
Comment (akwizgran) [reply app/src/main/java/org/onionshare/android/ui/settings/SettingsTorUi.kt:79]: I agree, better to do it on the UI thread. Thanks for explaining!

File: app/src/main/java/org/onionshare/android/ui/settings/SettingsUi.kt (hunk 4 above)
[the remark this answers is not in the document]
Comment (akwizgran) [reply app/src/main/java/org/onionshare/android/ui/settings/SettingsUi.kt]: Thanks!

File: app/src/main/java/org/onionshare/android/ui/settings/SettingsTorUi.kt (hunk 3 above)
Comment (akwizgran) [reply app/src/main/java/org/onionshare/android/ui/settings/SettingsTorUi.kt:239]: Sorry, I don't understand what it means for the text label to be disabled (therefore non-selectable?) when it's selected. How can it be non-selectable when it's selected, and why does this need to change based on whether it's selected?

Comment (grote) [reply app/src/main/java/org/onionshare/android/ui/settings/SettingsTorUi.kt:239]: I found it odd that I could re-select the already selected option, causing a re-composition and ui update with it. You can still do it when select the radio circle directly.

Comment (akwizgran) [reply app/src/main/java/org/onionshare/android/ui/settings/SettingsTorUi.kt:239]: Interesting, thanks for explaining!

**Pull request merged**